packages/dd-trace/src/llmobs/sdk.js

```diff
@@ -0,0 +1,448 @@
+    if (!this.enabled) {
+      logger.warn(
+        'LLMObs.submitEvaluation() called when LLMObs is not enabled. Evaluation metric data will not be sent.'
+      )
+      return
+    }
+
+    if (!this._config.llmobs.apiKey && !this._config.apiKey) {
+      logger.warn(
+        'DD_API_KEY is required for sending evaluation metrics. Evaluation metric data will not be sent.\n' +
+        'Ensure this configuration is set before running your application.'
+      )
+      return
+    }
+
+    const { traceId, spanId } = llmobsSpanContext
+    if (!traceId || !spanId) {
+      logger.warn(
+        'spanId and traceId must both be specified for the given evaluation metric to be submitted.'
+      )
+      return
+    }
+
+    const mlApp = options.mlApp || this._config.llmobs.mlApp
+    if (!mlApp) {
+      logger.warn('ML App name is required for sending evaluation metrics. Evaluation metric data will not be sent.')
+      return
+    }
+
+    const timestampMs = options.timestampMs || Date.now()
+    if (typeof timestampMs !== 'number' || timestampMs < 0) {
+      logger.warn('timestampMs must be a non-negative integer. Evaluation metric data will not be sent')
+      return
+    }
+
+    const { label, value, tags } = options
+    const metricType = options.metricType?.toLowerCase()
+    if (!label) {
+      logger.warn('label must be the specified name of the evaluation metric')
+      return
+    }
+    if (!metricType || !['categorical', 'score'].includes(metricType)) {
+      logger.warn('metricType must be one of "categorical" or "score"')
+      return
+    }
+
+    if (metricType === 'categorical' && typeof value !== 'string') {
+      logger.warn('value must be a string for a categorical metric.')
+      return
+    }
+    if (metricType === 'score' && typeof value !== 'number') {
+      logger.warn('value must be a number for a score metric.')
+      return
+    }
+
+    const evaluationTags = {
+      'dd-trace.version': tracerVersion,
+      ml_app: mlApp
+    }
+
+    if (tags) {
+      for (const key in tags) {
+        const tag = tags[key]
+        if (typeof tag === 'string') {
+          evaluationTags[key] = tag
+        } else if (typeof tag.toString === 'function') {
+          evaluationTags[key] = tag.toString()
+        } else {
+          logger.warn('Failed to parse tags. Tags for evaluation metrics must be strings')
+        }
+      }
+    }
+
+    const payload = {
+      span_id: spanId,
+      trace_id: traceId,
+      label,
+      metric_type: metricType,
+      ml_app: mlApp,
+      [`${metricType}_value`]: value,
+      timestamp_ms: timestampMs,
+      tags: Object.entries(evaluationTags).map(([key, value]) => `${key}:${value}`)
+    }
+
+    evalMetricAppendCh.publish(payload)
+  }
+
+  flush () {
+    if (!this.enabled) {
+      logger.warn('Flushing when LLMObs is disabled. No spans or evaluation metrics will be sent')
+      return
+    }
+
+    flushCh.publish()
+  }
+
+  active () {
```

Comment:
Do we need this to be public/user-facing? Ideally the less APIs we expose to users, the better (unless this is intended to be used by users). cc @Kyle-Verhoog